src/docs/json_rpc.md

```diff
@@ -56,150 +54,121 @@ connection and what happens on connection creation.
 It looks like this:
 
 ``` typescript
-import { MessageConnection } from "vscode-jsonrpc";
+import { Channel } from '../message-rpc/channel';
 
 export const ConnectionHandler = Symbol('ConnectionHandler');
 
 export interface ConnectionHandler {
     readonly path: string;
-    onConnection(connection: MessageConnection): void;
+    onConnection(connection: Channel): void;
 }
 ```
 
 ``` typescript
-import { ILoggerServer, ILoggerClient } from '../../application/common/logger-protocol';
+import { TaskClient, TaskServer, taskPath } from '../common/task-protocol';
 ```
 
-The logger-protocol.ts file contains the interfaces that the server and the
+The `task-protocol.ts` file contains the interfaces that the server and the
 client need to implement.
 
-The server here means the backend object that will be called over JSON-RPC
+The server here means the backend object that will be called over RPC
 and the client is a client object that can receive notifications from the
 backend object.
 
-I'll get more into that later.
+We will get more into that later.
 
 ``` typescript
     bind<ConnectionHandler>(ConnectionHandler).toDynamicValue(ctx => {
 ```
 
-Here a bit of magic happens, at first glance we're just saying here's an
-implementation of a ConnectionHandler.
-
-The magic here is that this ConnectionHandler type is bound to a
-ContributionProvider in messaging-module.ts
+Here a bit of magic happens, at first glance we're just saying here is an
+implementation of a `ConnectionHandler`.
 
-So as the MessagingContribution starts (onStart is called) it creates a
-websocket connection for all bound ConnectionHandlers.
-
-like so (from messaging-module.ts):
-
-``` typescript
-constructor( @inject(ContributionProvider) @named(ConnectionHandler) protected readonly handlers: ContributionProvider<ConnectionHandler>) {
```

Comment:
Not necessary?

Reply:
This snippet is outdated. The replacement in the current codebase would be the `MessagingContribution`.
However, this classe hides most of the relevant logic behind abstractions. I don't think a snippet of the new API is really helpful in this high level introduction . It's potentially more confusing than it helps.